### DISCUSSION TITLE
Sensor data update frequency

### DISCUSSION THREAD

**Discussion by priitka** [emporia-vue-local/esphome#89] [General]
First, many thanks to Flaviu for creating non-cloud firmware for Emporia Vue 2. Without it I wouldn't have bought it. 

I've noticed that readings are sent from esphome 2 times less frequently as sliding_window_moving_average send_every parameters specify. I've removed filters and dug a bit deeper into it and found out that readings from MCU arrive once in every 500 ms instead of 240 as commented in vue2.yaml file. Every 2nd EmporiaVueComponent::update() fails with "Ignoring sensor reading that is marked as read" notice. Perhaps MCU update frequency was changed in rev5 hardware or latest MCU firmware?

Not a big deal, as you can reduce sliding_window filter parameters.

**Comment by jcam** [comment]
Where did you get the 'ignoring sensor reading' log notice? was that coming through by default? I'd like to see whether either of my vue's do this too

**Reply by priitka** [reply]
You can see that in log when you set logger level to verbose
Another way to see it is to remove averaging filter and check how often you'll get update from Vue. It is 2x in second for my Vue

**Comment by spali** [comment]
Can confirm my rev5 does also return readings every 500ms when removing moving_avg filter.

**Comment by gupta-ash** [comment]
How do you find what rev HW you are on?

**Reply by spali** [reply]
Is written on the board somewhere.

**Comment by airdrummingfool** [comment]
If this is the case, is there a way to tell the underlying ESP component to poll less often? It looks like it is hard-coded here:
https://github.com/emporia-vue-local/esphome/blob/d7cc89375579c014a3e82b94f9c1cb1ad9ac3542/esphome/components/emporia_vue/emporia_vue.h#L43

Is there a way to override that value from the configuration YAML? For boards that only update every 500ms, it seems like changing that number would reduce the load on the ESP a bit.

**Reply by flaviut** [reply]
Pull requests accepted :)

I don't have time to do it myself unfortunately

**Reply by Gelisob** [reply]
> If this is the case, is there a way to tell the underlying ESP component to poll less often? It looks like it is hard-coded here:
> 
> https://github.com/emporia-vue-local/esphome/blob/d7cc89375579c014a3e82b94f9c1cb1ad9ac3542/esphome/components/emporia_vue/emporia_vue.h#L43
> 
> Is there a way to override that value from the configuration YAML? For boards that only update every 500ms, it seems like changing that number would reduce the load on the ESP a bit.

Hey, I see that I also have rev5. Did you test to figure out if that component file value change to 500 works?
Should change sliding_window_moving_average values too then..

**Reply by airdrummingfool** [reply]
I haven't tested any changes yet unfortunately. I haven't even confirmed that my rev5 only updates half as often as expected - I did watch the logs for a bit but only saw a handful of `Ignoring sensor reading that is marked as read` messages, so it's not clear to me if modifications are needed.

**Reply by spali** [reply]
Can't confirm the exact 500ms, but removing any filter changes the values of mine rev5 more or less twice a second. So assuming the 240 poll, we would be **_around_** the 500ms... but that value is an estimate. Could also just be 480ms or something around the that.

**Reply by airdrummingfool** [reply]
I did a bit of testing and can confirm that my rev5 gets 2 updates per second without any filters.

**Comment by airdrummingfool** [comment]
FYI to all on this discussion - it looks like [a recent pull request](https://github.com/emporia-vue-local/esphome/pull/188) added an undocumented parameter to allow configuration of the update interval. It also changed the default update interval to `0ms`, with the intention of making sure that the hardware doesn't miss any updates. 🚀
